pallet/account-migration/src/benchmarking.rs

```diff
@@ -0,0 +1,194 @@
+//
+// Darwinia is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
+
+// darwinia
+use crate::*;
+// substrate
+use frame_benchmarking::v2::{self, Linear};
+use frame_system::RawOrigin;
+use pallet_identity::{IdentityInfo, Judgement};
+use sp_std::prelude::*;
+
+#[v2::benchmarks]
+mod benchmarks {
+	// darwinia
+	use super::*;
+
+	fn preset_data<T>(from: &AccountId32)
+	where
+		T: Config,
+	{
+		let encoded_kton_id = KTON_ID.encode();
+		let kton = 1;
+
+		if let Some(mut asset_details) = migration::take_storage_value::<AssetDetails>(
+			b"Assets",
+			b"Asset",
+			&Blake2_128Concat::hash(&encoded_kton_id),
+		) {
+			asset_details.supply += kton;
+
+			migration::put_storage_value(
+				b"Assets",
+				b"Asset",
+				&Blake2_128Concat::hash(&encoded_kton_id),
+				asset_details,
+			);
+		}
+
+		<Accounts<T>>::insert(&from, AccountInfo::default());
+		<KtonAccounts<T>>::insert(
+			&from,
+			AssetAccount {
+				balance: kton,
+				is_frozen: Default::default(),
+				reason: ExistenceReason::Sufficient,
+				extra: Default::default(),
+			},
+		);
+		<Vestings<T>>::insert(
+			&from,
+			vec![
+				VestingInfo::new(Default::default(), Default::default(), Default::default());
+				<T as pallet_vesting::Config>::MAX_VESTING_SCHEDULES as usize
+			],
+		);
+		<Identities<T>>::insert(
+			&from,
+			Registration {
+				judgements: BoundedVec::truncate_from(vec![
+					(Default::default(), Judgement::Unknown);
+					<T as pallet_identity::Config>::MaxRegistrars::get()
+						as usize
+				]),
+				deposit: Default::default(),
+				info: IdentityInfo {
+					additional: BoundedVec::truncate_from(vec![
+						(
+							Default::default(),
+							Default::default()
+						);
+						<T as pallet_identity::Config>::MaxAdditionalFields::get()
+							as usize
+					]),
+					display: Default::default(),
+					legal: Default::default(),
+					web: Default::default(),
+					riot: Default::default(),
+					email: Default::default(),
+					pgp_fingerprint: Some(Default::default()),
+					image: Default::default(),
+					twitter: Default::default(),
+				},
+			},
+		);
+		<Ledgers<T>>::insert(
+			&from,
+			Ledger {
+				staked_ring: Default::default(),
+				staked_kton: kton,
+				staked_deposits: BoundedVec::truncate_from(vec![
+					Default::default();
+					<T as darwinia_deposit::Config>::MaxDeposits::get()
+						as usize
+				]),
+				unstaking_ring: BoundedVec::truncate_from(vec![
+					(
+						Default::default(),
+						Default::default()
+					);
+					<T as darwinia_staking::Config>::MaxUnstakings::get()
+						as usize
+				]),
+				unstaking_kton: BoundedVec::truncate_from(vec![
+					(
+						Default::default(),
```

Comment:
And this.

Reply:
What do you mean "and this"?

Reply:
See https://github.com/darwinia-network/darwinia/pull/1029#discussion_r1139555703. The current version of benchmark does not cover the balance transfer and asset transafer cases.

Reply:
See `staked_kton += 1;`. Assets transfer is calculated.